src/remover.rs

```diff
@@ -8,8 +8,11 @@ pub fn remove_dep(
 ) -> Result<String> {
     let remove_dep = remove_dep_opt.context("error: expected dep to remove")?;
 
-    let range_to_remove = find_remove_dep(deps_list, &remove_dep)
-        .context("error: could not find dependency to remove")?;
+    let search = find_remove_dep(deps_list, &remove_dep);
+    if search.is_err() {
+        return Ok(contents.to_string());
```

Comment:
can we change the function to return `Result<Option<String>>` and returning `None` here instead? that way we can avoid updating the mtime of the dirent and won't trigger nix rebuilds :)

Reply:
We [don't write the file here](https://github.com/replit/nix-editor/blob/671ea01ad18c149f46330426a229c4f6c616467a/src/main.rs#L289) if the contents are the same.

Reply:
I confirmed the mtime is not updating myself. I agree that you're way will be better, will add it next time I dive into the remover code.